### PR TITLE
glfs: add gfapi version check

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <xiubli@redhat.com>

### CODE REVIEW

**Comment by lxbsz** [comment]
@mikechristie @pkalever @amarts 
Please review. Thanks.

**Comment by amarts** [comment]
@nixpanic @ShyamsundarR @kalebskeithley ^^

File: glfs.c [hunk 1]
```diff
@@ -566,7 +567,13 @@ static void tcmu_glfs_close(struct tcmu_device *dev)
 	free(gfsp);
 }
 
+#if GFAPI_VERSION < 741
 static void glfs_async_cbk(glfs_fd_t *fd, ssize_t ret, void *data)
+#elif GFAPI_VERSION < 760
+static void glfs_async_cbk(glfs_fd_t *fd, ssize_t ret, struct stat *prestat, struct stat *poststat, void *data)
```
Comment (nixpanic) [inline glfs.c]: I do not think this variation exists? GlusterFS 4.1 and 5 have

```
typedef void (*glfs_io_cbk)(glfs_fd_t *fd, ssize_t ret, void *data);
````

Comment (lxbsz) [reply glfs.c]: yeah, also the same, just checked this by using the 'git tag --contains SHA', which has been reverted in all the release branches.

Will update it.

**Comment by lxbsz** [comment]
@nixpanic 
Update done, please review and thanks.

File: glfs.c (hunk 1 above)
Comment (lxbsz) [reply glfs.c]:
```
commit 384562b294e9a7847403961e878a4daa0fff33eb
Author: Kinglong Mee <kinglongmee@gmail.com>
Date:   Thu Feb 1 11:20:37 2018 +0800

    gfapi: return pre/post attributes at callback for glfs api
    
    Updates: #389
    Change-Id: Ic71632722effe4b8855d5de3e65688efd9afe1e3
    Signed-off-by: Kinglong Mee <mijinlong@open-fs.com>

diff --git a/api/src/glfs.h b/api/src/glfs.h
index e2f2a2c..c8b261e 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -499,7 +499,9 @@ int glfs_set_xlator_option (glfs_t *fs, const char *xlator, const char *key,
   in a common callback function.
 */
 
-typedef void (*glfs_io_cbk) (glfs_fd_t *fd, ssize_t ret, void *data);
+typedef void (*glfs_io_cbk34) (glfs_fd_t *fd, ssize_t ret, void *data);
+typedef void (*glfs_io_cbk) (glfs_fd_t *fd, ssize_t ret, struct stat *prestat,
+                             struct stat *poststat, void *data);
 
 // glfs_{read,write}[_async]
 

```

**Comment by lxbsz** [comment]
> Hey @lxbsz
> 
> I am getting this error:
> 
> > glfs.c:570:19: error: operator '<' has no left operand
> > #if GFAPI_VERSION < 760
> > ^
> 
> when trying to build this. Do I need a specific gluster lib version?

Hi Mike, I fixed this in https://github.com/open-iscsi/tcmu-runner/pull/541https://github.com/open-iscsi/tcmu-runner/pull/541.

I should allow us to build this in any version here.

Thanks.

**Pull request merged**